src/Symfony/Bridge/Doctrine/DependencyInjection/CompilerPass/RegisterDatePointTypePass.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Doctrine\DependencyInjection\CompilerPass;
+
+use Symfony\Bridge\Doctrine\Types\DatePointType;
+use Symfony\Component\Clock\DatePoint;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+final class RegisterDatePointTypePass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container): void
+    {
+        if (!class_exists(DatePoint::class)) {
+            return;
+        }
+
+        if (!$container->hasParameter('doctrine.dbal.connection_factory.types')) {
+            return;
+        }
+
+        $typeDefinition = $container->getParameter('doctrine.dbal.connection_factory.types');
+
+        if (!isset($typeDefinition['date_point'])) {
```

Comment:
I don't get why psalm thinks `$typeDefinition` is an `UnitEnum`.

We can add a type check to be safe in case the parameter contains something else.

```suggestion
        if (is_array($typeDefinition) && !isset($typeDefinition['date_point'])) {
```

Reply:
It's weird, since I copied the test from the Uuid/Ulid one, and this portion of code is the same (except the name of course). I even tried to check if there's some kind of baseline, but unfortunately I'm not very familiar with psalm (I use phpstan in all my projects)

Reply:
I tried your suggestion and it worked. Thanks!